src/client/dfuse/pil4dfs/hook.c

```diff
@@ -583,16 +675,16 @@ allocate_memory_block_for_patches(void)
 			}
 
 			if (iSeg < 0) {
-				printf("Something wrong! The address you queried is not\n"
-				       "inside any module! Quit\n");
+				D_FATAL("Something wrong! The address you queried is not "
+					"inside any module! Quit\n");
 				exit(1);
 			}
 			p_Alloc = (void *)(addr_max[iSeg]);
 
 			if (iSeg < (num_seg - 1)) {
 				if ((addr_min[iSeg + 1] - addr_max[iSeg]) < MIN_MEM_SIZE) {
-					printf("Only %" PRIu64 " bytes available.\nQuit\n",
-					       addr_min[iSeg + 1] - addr_max[iSeg]);
+					D_FATAL("Only %" PRIu64 " bytes available.\nQuit\n",
```

Comment:
No need for the extra `\nQuit` here.

As an error message this is not helpful to anyone.  Imagine a user of `pil4fds` who sees this string when they try to start their application, how would they be expected to resolve the issue?  See my earlier comments about having this as a D_ERROR and then a common D_FATAL which was more descriptive before the program exited.
```suggestion
					D_FATAL("Only %" PRIu64 " bytes available.\n",
```